qiskit/algorithms/optimizers/suffix_averaging.py

```diff
@@ -0,0 +1,144 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Suffix Averaging optimier"""
+
+import warnings
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import numpy as np
+
+from .gradient_descent import GradientDescent
+from .nft import NFT
+from .optimizer import POINT, Optimizer, OptimizerResult
+from .qnspsa import QNSPSA
+from .spsa import SPSA
+
+
+class SuffixAveragingOptimizer(Optimizer):
+    r"""The suffix averaging optimizer.
+
+    Given a sequence of circuit parameters of parameterized quantum circuit
+    obtained from :math:`T` iterations, :math:`\{\vec{\theta}^{(t)}\}_{t=1}^{t=T}`,
+    the SuffixAveragingOptimizer returns the averaged point of the last n_params_suffix
+    points,
+
+    .. math::
+
+        \overline{\vec{\theta}} = \frac{1}{n_params_suffix} \sum_{t=T-n_params_suffix-1}^{T}
+        \vec{\theta}^{(t)}.
+
+    References:
+        [1] S. Tamiya and H. Yamasaki. 2021.
+        Stochastic Gradient Line Bayesian Optimization
+        for Efficient Noise-Robust Optimization of Parameterized Quantum Circuits.
+        arXiv preprint arXiv:2111.07952.
+    """
+
+    def __init__(self, optimizer: Optimizer, n_params_suffix: int = 50) -> None:
```

Comment:
50 seems pretty random, is there an intuition for that? With many default steps being like 100 or 300, it seems also a bit small. We could probably make this a required argument here, since this is really important!